services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java

```diff
@@ -253,27 +254,31 @@ private void initAttestationPropagationManager() {
 
   public void initP2PNetwork() {
     LOG.debug("BeaconChainController.initP2PNetwork()");
-    if ("mock".equals(config.getNetworkMode())) {
+    if (!config.isP2pEnabled()) {
       this.p2pNetwork = new MockP2PNetwork<>(eventBus);
-    } else if ("jvmlibp2p".equals(config.getNetworkMode())) {
-      Bytes bytes = Bytes.fromHexString(config.getInteropPrivateKey());
+    } else {
+      final Bytes bytes;
+      try {
+        bytes = Bytes.fromHexString(Files.readString(Paths.get(config.getP2pPrivateKeyFile())));
+      } catch (IOException e) {
+        throw new RuntimeException(
+            "p2p private key file not found - " + config.getP2pPrivateKeyFile());
+      }
       PrivKey pk =
           bytes.isEmpty()
```

Comment:
Should we assert that bytes is not empty and cut the private key generation logic here?

Reply:
I don't think you need to assert here. `Bytes.fromHexString(..)` already asserts the input is not null.

Yes keygen logic is not necessary!

Reply:
This is probably a bad thing because I am changing the execution paths - I think a key should be generated if the file path is not supplied - but an exception should be thrown if an invalid file path is supplied.
I will update to reflect this.